### DISCUSSION TITLE
export data from jatos server

### DISCUSSION THREAD

**Discussion by m16i** [jspsych/jsPsych#1239] [Q&A] [answered]
Hello everyone, 
I have little experience in web programing and development and have followed the online tutorials provided by "Moving research online" for which I am very grateful. 
I would like to use jPsych withing jatos. When I trying to use the examples written in jsPsych on the jatos site I have problems with data exporting: more specifically, when exporting data from multiple subjects I have difficulty accessing the files due to invalid json format. In addition I noticed that there are no default variables generated by jatos which would help to create data with server side generated codes allowing for an association between single trial and session. Is there a way to solve both problems? I would be grateful for you feedback on the above and if you could suggest any relevant tutorials on the subject. 

Thank you in advance, 

Michele M

**Comment by jodeleeuw** [comment]
Hi Michele,

You might have more luck asking the JATOS folks about this. It sounds like the issues are with JATOS functionality and not jsPsych. Let us know if it turns out that the problems are caused by jsPsych so we can address it!

**Comment by amunn** [comment, marked as answer]
`jatos.js` provides a number of ID variables you can find [here](https://www.jatos.org/jatos.js-Reference.html#id-variables) which you can add using `jsPsych.data.addProperties` in the `on_finish` function of `initJsPsych`. (Unless I've misunderstood the second part of your question). 

But the exporting multiple participants data in JSON format does seem to be a JATOS problem. I've asked about it [here](https://forum.cogsci.nl/discussion/8081/jatos-export-results-in-json-format/p1?new=1).  In the meantime, I've cooked up a python script to convert the invalid JATOS combined output to valid JSON that can be imported e.g. using the `jsonlite` package in R.

https://github.com/amunn/fixjatos

Comments welcome. 

(It seems that Michele found a different solution [here](https://github.com/JATOS/JATOS/issues/213))
